slicedimage/io.py

```diff
@@ -188,8 +188,9 @@ def parse(self, json_doc, baseurl):
                         sha256=tile_doc.get(TileKeys.SHA256, None),
                         extras=tile_doc.get(TileKeys.EXTRAS, None),
                     )
+                    seekable = True if tile_format == ImageFormat.NUMPY else False
                     tile.set_source_fh_contextmanager(
-                        backend.read_file_handle_callable(name), tile_format)
+                        backend.read_file_handle_callable(name, seekable=seekable), tile_format)
```

Comment:
I'd love it if we could add a block that uses 
```
tile_format = ImageFormat.NUMPY
with pytest.raises(ExpectedError):
    backend.read_file_handle_callable(name, seekable=True, tile_format)
```

It verifies that all this code is necessary, and if numpy ever changed for the better we'd get an indication that we can go back and treat everything the same. 

I could imagine either adding a test for this small piece of the code, or using a mock to patch the `test_numpy` test wherein you replace `backend.read_file_handle_callable` with a `seekable=False` argument.

Reply:
Done.  see `test_numpy_requires_seekable`